src/AutoMerge/new-package.jl

```diff
@@ -79,170 +79,70 @@ function pull_request_build(data::GitHubAutoMergeData, ::NewPackage)::Nothing
-        m6 = ""
-    else
-        # g6, m6 = check!(G6)
-        g6 = true
-        m6 = ""
+    if !this_pr_can_use_special_jll_exceptions
+        # push!(guidelines,
+        #       guideline_standard_initial_version_number) # 6
     end
 
-    G7 = guideline_repo_url_requirement
-    # g7, m7 = check!(G7)
-    g7 = true
-    m7 = ""
-
-    G8 = guideline_compat_for_all_deps
-    g8, m8 = check!(G8)
+    push!(guidelines,
+          # guideline_repo_url_requirement, #7
+          guideline_compat_for_all_deps) #8
 
-    G9 = guideline_allowed_jll_nonrecursive_dependencies
     if this_is_jll_package
-        g9, m9 = check!(G9)
-    else
-        g9 = true
-        m9 = ""
+        push!(guidelines,
+              guideline_allowed_jll_nonrecursive_dependencies) # 9
     end
 
-    G10 = guideline_distance_check
-    g10, m10 = check!(G10)
+    push!(guidelines,
+          guideline_distance_check, # 10
+          guideline_name_ascii) # 11
 
-    G11 = guideline_name_ascii
-    g11, m11 = check!(G11)
+    for guideline in guidelines
+        check!(guideline)
+        @info(guideline.info,
+              meets_this_guideline = passed(guideline),
+              message = message(guideline))
+    end
 
-    @info(G0.info,
-          meets_this_guideline = g0,
-          message = m0)
-    @info(G1.info,
-          meets_this_guideline = g1,
-          message = m1)
-    @info(G2.info,
-          meets_this_guideline = g2,
-          message = m2)
-    @info(G3.info,
-          meets_this_guideline = g3,
-          message = m3)
-    @info(G4.info,
-          meets_this_guideline = g4,
-          message = m4)
-    @info(G5.info,
-          meets_this_guideline = g5,
-          message = m5)
-    @info(G6.info,
-          meets_this_guideline = g6,
-          message = m6)
-    @info(G7.info,
-          meets_this_guideline = g7,
-          message = m7)
-    @info(G8.info,
-          meets_this_guideline = g8,
-          message = m8)
-    @info(G9.info,
-          meets_this_guideline = g9,
-          message = m9)
-    @info(G10.info,
-          meets_this_guideline = g10,
-          message = m10)
-    @info(G11.info,
-          meets_this_guideline = g11,
-          message = m11)
-    g0through11 = Bool[g0,
-                      g1,
-                      g2,
-                      g3,
-                      g4,
-                      g5,
-                      g6,
-                      g7,
-                      g8,
-                      g9,
-                      g10,
-                      g11]
-    if !all(g0through11)
+    if !all(passed, guidelines)
         update_status(data;
                       state = "failure",
                       context = "automerge/decision",
                       description = "New package. Failed.")
     end
 
-    G12 = guideline_version_can_be_pkg_added
-    g12, m12 = check!(G12)
-    @info(G12.info,
-          meets_this_guideline = g12,
-          message = m12)
+    push!(guidelines,
```

Comment:
I think it might be nicer to have two separate vectors, like `guidelines_stage_1` and `guidelines_stage_2`, where here we are starting `stage_2`. That would help that if we add a "stage 2" guideline, we don't have to ensure we update the indexing `guidelines[end-1:end]` below at the same time (since instead at the second check, we just loop over `guidelines_stage_2`. Then we could `vcat` them into just `guidelines` later if that's helpful.

Reply:
Agreed that this isn't so nice and also part of note 2. My idea was to have a single input queue but inject `:update_status` or something to that effect between the guidelines, when it's time to update the status. (If that's needed at all, I'm not clear  on the timing aspects of these checks.)

Reply:
You can think of the checks in two parts: 
1. Everything before part 2
2. Part 2, which is `Pkg.add` and `import`.

The part 2 checks can take a long time depending on the package. If the part 1 checks fail, then we update the AutoMerge/decision status with a fail before we run part 2, so that people can get quick feedback even while they wait for the part 2 checks to finish.